### PR TITLE
Updated AccountState() to use address.Address

### LINKED ISSUE #2775: [account] AccountState() to use address.Address
**What would you like to be added**:
similar to #2771, change to `func AccountState(sr protocol.StateReader, addr address.Address)`
and also `AccountStateWithHeight`

**Why is this needed**:
see #2771

### PR DESCRIPTION
**Pull request by ashishsnigam**
Fixed #2775

### CODE REVIEW

File: action/protocol/account/util/util.go [hunk 1]
```diff
@@ -86,18 +86,24 @@ func Recorded(sr protocol.StateReader, addr address.Address) (bool, error) {
...
-	a, _, err := AccountStateWithHeight(sr, encodedAddr)
+// AccountState returns the confirmed account state on the chain by address.Address
+func AccountState(sr protocol.StateReader, addr address.Address) (*state.Account, error) {
+	a, _, err := AccountStateWithHeight(sr, addr)
 	return a, err
 }
 
-// AccountStateWithHeight returns the confirmed account state on the chain with what height the state is read from.
-func AccountStateWithHeight(sr protocol.StateReader, encodedAddr string) (*state.Account, uint64, error) {
+// AccountStateByHash160 returns the confirmed account state on the chain by 20-byte address
```
Comment (dustinxie) [inline action/protocol/account/util/util.go:95]: delete this func, we change the input from `string` to `address.Address`

File: action/protocol/account/transfer_test.go [hunk 2]
```diff
@@ -113,9 +113,9 @@ func TestProtocol_HandleTransfer(t *testing.T) {
 			GasLimit:    testutil.TestGasLimit,
 		})
 
-		sender, err := accountutil.AccountState(sm, v.caller.String())
+		sender, err := accountutil.AccountState(sm, v.caller)
 		require.NoError(err)
-		recipient, err := accountutil.AccountState(sm, v.recipient)
+		recipient, err := accountutil.AccountStateByHash160(sm, v.recipient)
```
Comment (dustinxie) [inline action/protocol/account/transfer_test.go:118]: change `v.recipient` to address.Address

Comment (dustinxie) [reply action/protocol/account/transfer_test.go:118]: do the same for other tests

File: state/factory/workingset.go [hunk 3]
```diff
@@ -309,7 +309,7 @@ func (ws *workingSet) validateNonce(blk *block.Block) error {
 	}
 	// Verify each account's Nonce
 	for srcAddr, receivedNonces := range accountNonceMap {
-		confirmedState, err := accountutil.AccountState(ws, srcAddr)
+		confirmedState, err := accountutil.AccountStateByHash160(ws, srcAddr)
```
Comment (dustinxie) [inline state/factory/workingset.go:312]: change `accountNonceMap` to map[address.Address][]uint64

File: blockchain/integrity/integrity_test.go [hunk 4]
```diff
@@ -1261,12 +1261,14 @@ func TestLoadBlockchainfromDB(t *testing.T) {
 		fmt.Printf("Cannot add block 3 again: %v\n", err)
 
 		// invalid address returns error
-		act, err := accountutil.AccountState(sf, "")
```
Comment (ashishsnigam) [inline blockchain/integrity/integrity_test.go:1264]: @dustinxie what to put here to make this test pass ?

File: action/protocol/execution/protocol_test.go [hunk 5]
```diff
@@ -267,7 +267,11 @@ func runExecutions(
 		var ok bool
 		executor := ecfg.Executor().String()
```
Comment (dustinxie) [inline action/protocol/execution/protocol_test.go:268]: `executor := ecfg.Executor()`
then can use `accountutil.AccountState(sf, executorAddr)` below

File: api/api.go [hunk 6]
```diff
@@ -453,11 +453,18 @@ func (api *Server) ReadContract(ctx context.Context, in *iotexapi.ReadContractRe
...
-	state, err := accountutil.AccountStateByHash160(api.sf, in.CallerAddress)
+	addr, err := address.FromString(in.CallerAddress)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+	state, err := accountutil.AccountState(api.sf, addr)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+	callerAddr, err := address.FromString(in.CallerAddress)
```
Comment (dustinxie) [inline api/api.go:464]: same code as line 456 above

File: e2etest/bigint_test.go [hunk 7]
```diff
@@ -8,6 +8,7 @@ package e2etest
 
 import (
 	"context"
+	"github.com/iotexproject/iotex-address/address"
```
Comment (dustinxie) [inline e2etest/bigint_test.go:11]: move to line 16

File: e2etest/local_transfer_test.go [hunk 8]
```diff
@@ -9,6 +9,7 @@ package e2etest
 import (
 	"context"
 	"fmt"
+	"github.com/iotexproject/iotex-address/address"
```
Comment (dustinxie) [inline e2etest/local_transfer_test.go:12]: move to line 18 below

File: state/factory/workingset.go [hunk 9]
```diff
@@ -8,6 +8,7 @@ package factory
 
 import (
 	"context"
+	"github.com/iotexproject/iotex-address/address"
```
Comment (dustinxie) [inline state/factory/workingset.go:11]: move to line 14 below

File: actpool/actpool_test.go [hunk 10]
```diff
@@ -9,6 +9,7 @@ package actpool
 import (
 	"bytes"
 	"context"
+	"github.com/iotexproject/iotex-address/address"
```
Comment (dustinxie) [inline actpool/actpool_test.go:12]: move to line 23 below

Comment (dustinxie) [reply actpool/actpool_test.go:12]: also move line 18-21 together with other `/iotex-core`

File: blockchain/integrity/integrity_test.go (hunk 4 above)
Comment (dustinxie) [reply blockchain/integrity/integrity_test.go:1264]: `_, err = address.FromString("")`

File: actpool/actpool_test.go (hunk 10 above)
Comment (ashishsnigam) [reply actpool/actpool_test.go:12]: done..imports sorted as per go-fmt

File: state/factory/workingset.go (hunk 9 above)
Comment (ashishsnigam) [reply state/factory/workingset.go:11]: done

File: e2etest/bigint_test.go (hunk 7 above)
Comment (ashishsnigam) [reply e2etest/bigint_test.go:11]: done

File: api/api.go (hunk 6 above)
Comment (ashishsnigam) [reply api/api.go:464]: done.. removed duplicate code

File: action/protocol/execution/protocol_test.go (hunk 5 above)
Comment (ashishsnigam) [reply action/protocol/execution/protocol_test.go:268]: done

File: blockchain/integrity/integrity_test.go (hunk 4 above)
Comment (ashishsnigam) [reply blockchain/integrity/integrity_test.go:1264]: done

File: state/factory/workingset.go (hunk 3 above)
Comment (ashishsnigam) [reply state/factory/workingset.go]: already done

File: action/protocol/account/transfer_test.go (hunk 2 above)
Comment (ashishsnigam) [reply action/protocol/account/transfer_test.go]: done

File: action/protocol/account/util/util.go (hunk 1 above)
Comment (ashishsnigam) [reply action/protocol/account/util/util.go]: done

**Comment by ashishsnigam** [comment]
Hi @dustinxie ,  I incorporated all your review comments. Still 3 tests are failing in this branch out of which 1 is also failing on master branch.

--- FAIL: TestLocalTransfer (3.84s).      (Also failing on master branch)
    local_transfer_test.go:362:
                Error Trace:    local_transfer_test.go:362
                Error:          Received unexpected error:
                                context deadline exceeded
                Test:           TestLocalTransfer

--- FAIL: TestStartExistingBlockchain (3.86s)
    local_test.go:574:
                Error Trace:    local_test.go:574
                Error:          Received unexpected error:
                                invalid nonce


--- FAIL: TestLocalSync (64.54s)
    local_test.go:508:
                Error Trace:    local_test.go:508
                Error:          Received unexpected error:
                                timed out

Need help to fix those tests.

File: action/protocol/generic_validator.go [hunk 11]
```diff
@@ -8,6 +8,7 @@ package protocol
 
 import (
 	"context"
+	"github.com/iotexproject/iotex-address/address"
```
Comment (dustinxie) [inline action/protocol/generic_validator.go:11]: move to after line 14

Comment (ashishsnigam) [reply action/protocol/generic_validator.go:11]: done

**Comment by ashishsnigam** [comment]
opened a new PR #2803

**Pull request closed without merging**